daemon/api_snaps.go

```diff
@@ -355,6 +359,13 @@ func (inst *snapInstruction) validate() error {
 		}
 	}
 
+	if inst.Unaliased && inst.Prefer {
+		return errors.New(`cannot use unaliased and prefer at the same time`)
```

Comment:
Consider using `specify` instead of `cannot use` similar [to this](https://github.com/snapcore/snapd/pull/12993/files#diff-1857121b555d20efa2d46ca1e69c987d21d1bd12757cd5023f82d087820251acR177)